tests/components/konnected/test_init.py

```diff
@@ -0,0 +1,601 @@
+                "2": "Binary Sensor",
+                "3": "Binary Sensor",
+                "4": "Digital Sensor",
+                "5": "Digital Sensor",
+                "6": "Switchable Output",
+                "out": "Switchable Output",
+            },
+            "binary_sensors": [
+                {"zone": "1", "type": "door"},
+                {"zone": "2", "type": "window", "name": "winder", "inverse": True},
+                {"zone": "3", "type": "door"},
+            ],
+            "sensors": [
+                {"zone": "4", "type": "dht"},
+                {"zone": "5", "type": "ds18b20", "name": "temper"},
+            ],
+            "switches": [
+                {
+                    "zone": "out",
+                    "name": "switcher",
+                    "activation": "low",
+                    "momentary": 50,
+                    "pause": 100,
+                    "repeat": 4,
+                },
+                {"zone": "6"},
+            ],
+        }
+    )
+
+    entry = MockConfigEntry(
+        domain="konnected",
+        title="Konnected Alarm Panel",
+        data=device_config,
+        options=device_options,
+    )
+    entry.add_to_hass(hass)
+
+    assert (
+        await async_setup_component(
+            hass,
+            konnected.DOMAIN,
+            {konnected.DOMAIN: {konnected.CONF_ACCESS_TOKEN: "globaltoken"}},
+        )
+        is True
+    )
+
+    client = await aiohttp_client(hass.http.app)
+
+    # Test the get endpoint for switch status polling
+    resp = await client.get("/api/konnected")
+    assert resp.status == 404  # no device provided
+
+    resp = await client.get("/api/konnected/223344556677")
+    assert resp.status == 404  # unknown device provided
+
+    resp = await client.get("/api/konnected/device/112233445566")
+    assert resp.status == 404  # no zone provided
+    result = await resp.json()
+    assert result == {"message": "Switch on zone or pin unknown not configured"}
+
+    resp = await client.get("/api/konnected/device/112233445566?zone=8")
+    assert resp.status == 404  # invalid zone
+    result = await resp.json()
+    assert result == {"message": "Switch on zone or pin 8 not configured"}
+
+    resp = await client.get("/api/konnected/device/112233445566?pin=12")
+    assert resp.status == 404  # invalid pin
+    result = await resp.json()
+    assert result == {"message": "Switch on zone or pin 12 not configured"}
+
+    resp = await client.get("/api/konnected/device/112233445566?zone=out")
+    assert resp.status == 200
+    result = await resp.json()
+    assert result == {"state": 1, "zone": "out"}
+
+    resp = await client.get("/api/konnected/device/112233445566?pin=8")
+    assert resp.status == 200
+    result = await resp.json()
+    assert result == {"state": 1, "pin": "8"}
+
+    # Test the post endpoint for sensor updates
+    resp = await client.post("/api/konnected/device", json={"zone": "1", "state": 1})
+    assert resp.status == 404
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4986898)